README.md

```diff
@@ -92,31 +111,49 @@ After a Key-Value pair is inserted into OakMap, it is kept in a serialized (buff
 Thus, while searching through the map, OakMap might compare between keys in their Object and Serialized modes. OakMap provides the following interface for such a comparator:
 ```java
 public interface OakComparator<K> {
-
   int compareKeys(K key1, K key2);
 
-  int compareSerializedKeys(ByteBuffer serializedKey1, ByteBuffer serializedKey2);
+  int compareSerializedKeys(OakReadBuffer serializedKey1, OakReadBuffer serializedKey2);
 
-  int compareSerializedKeyAndKey(ByteBuffer serializedKey, K key);
+  int compareSerializedKeyAndKey(OakReadBuffer serializedKey1, K key2);
 }
 ```
 
-This is how to create the comparator class in your code:
+*Note 1*: Oak use dedicated objects to access off-heap memory: `OakReadBuffer` and `OakWriteBuffer`.
+See [Oak Buffers](#oak-buffers) for more information.
+
+*Note 2*: `OakReadBuffer` and `OakWriteBuffer` should not be stored for future use.
+They are valid only in the context of these methods (`compareKeys()`/`compareSerializedKeys()/compareSerializedKeyAndKey()`).
+Using these buffers outside their intended context may yield unpredicted results.
+
+For example, the implementation of key serializer for an application that use integer as keys might like this:
 
 ```java
-public class OakKeyComparatorImplementation implements OakComparator<K>
-{...}
+public class MyAppKeyComparator implements OakComparator<Integer>
+{
+  int compareKeys(Integer key1, Integer key2) {
+    return key1 - key2;
+  }
+
+  int compareSerializedKeys(OakReadBuffer serializedKey1, OakReadBuffer serializedKey2) {
+    return serializedKey1.getInt(0) - serializedKey2.getInt(0); 
+  }
+
+  int compareSerializedKeyAndKey(OakReadBuffer serializedKey1, Integer key2) {
+    return serializedKey1.getInt(0) - key2;
+  }
+}
 ```
 
 ### Builder
 We provide an example how to create OakMapBuilder and OakMap. For a more comprehensive code example please refer to the [Usage](#usage) section.
```

Comment:
I see that you updated Usage section as well which is great. How can I comment on that content? Is it going to be just copied to main repository as it is not part of PR, I think...?

Reply:
I thought you updated the wiki